otlp/common.go

```diff
@@ -48,7 +48,8 @@ const (
 const fieldSizeMax = math.MaxUint16
 
 var (
-	legacyApiKeyPattern = regexp.MustCompile("^[0-9a-f]{32}$")
+	legacyApiKeyPattern     = regexp.MustCompile("^[0-9a-f]*$")
+	classicIngestKeyPattern = regexp.MustCompile("^hc[a-z]ic_[0-9a-z]*$")
```

Comment:
happy to adjust naming here

Reply:
"Legacy" is a legacy term for "Classic" before the name "Classic" was settled on. I'm 👍🏻 with a whole-sale switchover to "classic" in code we update, especially as we add public functions/methods.

Reply:
I'll update that public method name then 👍🏻